mobile/mysterium/entrypoint.go

```diff
@@ -212,6 +207,35 @@ func (mb *MobileNode) GetProposal(req *GetProposalRequest) ([]byte, error) {
 	return proposal, nil
 }
 
+// ProposalChangeCallback represents proposal callback.
+type ProposalChangeCallback interface {
+	OnChange(proposal []byte)
+}
+
+// RegisterProposalAddedCallback registers callback which is called on newly announced proposals
+func (mb *MobileNode) RegisterProposalAddedCallback(cb ProposalChangeCallback) {
+	_ = mb.eventBus.SubscribeAsync(discovery.ProposalAddedEventTopic, func(proposal market.ServiceProposal) {
+		proposalPayload, _ := mb.proposalsManager.mapToProposalResponse(&proposal)
+		cb.OnChange(proposalPayload)
```

Comment:
Sad news are that mapToProposalResponse is not adding quality level data. Before it was used only for single GetProposal which doesn't need quality data. But now these proposals need to have such data as they will update actual UI. See `func (m *proposalsManager) addQualityData(proposals []*proposal)` for usage how it was done. Probably quality metrics need to be cached on initial GetProposals etc. to be accessible for each event.

Reply:
Looks like need to addressed in another PR

Reply:
Could be possible addressed in https://github.com/mysteriumnetwork/node/issues/1524